In [3]:
import pandas as pd

inventory = []
num_items = int(input("Greetings, brave adventurer! How many treasures have you unearthed today?: "))

for _ in range(num_items):
    name = input("What is this magnificent item's name?: ")
    strength = int(input("How powerful is this item? (Give me a number!): "))
    rarity = int(input("Rate its rarity for me! (1=Common, 2=Uncommon, 3=Rare, 4=Epic, 5=Legendary): "))
    weight = float(input("How much does this item weigh? (Enter a number, please!): "))
    value = int(input("What price would this item fetch? (Numbers, my friend!): "))
    inventory.append({"name": name, "strength": strength, "rarity": rarity, "weight": weight, "value": value})

df_inventory = pd.DataFrame(inventory)

rarity_labels = {1: "Common", 2: "Uncommon", 3: "Rare", 4: "Epic", 5: "Legendary"}
df_inventory["rarity"] = df_inventory["rarity"].map(rarity_labels)

criteria = input("How would you like it sorted? (name/strength/rarity/weight/value): ").lower()
order = input("Ascending or Descending order? (type 'ascending' or 'descending') ").lower()
ascending = True if order == "ascending" else False

def bubble_sort(df, key, ascending=True):
    data = df.to_dict('records')  
    n = len(data)
    for i in range(n):
        for j in range(0, n - i - 1):
            if (ascending and data[j][key] > data[j + 1][key]) or (not ascending and data[j][key] < data[j + 1][key]):
                data[j], data[j + 1] = data[j + 1], data[j]
    return pd.DataFrame(data)


if criteria in df_inventory.columns:
    df_inventory = bubble_sort(df_inventory, criteria, ascending)
else:
    print("Invalid sorting criteria! Choose from 'name', 'strength', 'rarity', 'weight', or 'value'.")

df_inventory.index = range(1, len(df_inventory) + 1)

print("\nYour Inventory:")
print(df_inventory)



Your Inventory:
                                name  strength     rarity  weight  value
1            Elixir of Eternal Youth         0  Legendary    0.20  30000
2                            Mjolnir       302  Legendary   88.00  16000
3                          Excalibur       235  Legendary   40.00  15000
4                       Ethereal Bow        60  Legendary    3.20   7000
5                 Dragon Scale Armor        90  Legendary   15.00   5000
6                    Phoenix Feather         0  Legendary    0.05   5000
7                       Mystic Staff        80  Legendary    6.00   3000
8                  Whispering Amulet        20       Epic    0.50   3000
9   Golden Compass of the Lost Isles        15       Rare    1.30   2700
10                      Golden Crown        15       Epic    3.00   2500
11                   Frostbite Sword        85       Epic    8.00   2200
12                     Enchanted Bow        60       Epic    5.00   2000
13                   Sword of Flam

20
Bow
28
1
5
120
Sword of Flames
75
4
7.5
1500
Potion of Healing
0
2
0.5
50
Shield of Light
50
3
10
1200
Enchanted Bow
60
4
5
2000
Iron Dagger
25
1
2
100
Dragon Scale Armor
90
5
15
5000
Ring of Protection
20
3
0.2
800
Mystic Staff
80
5
6
3000
Golden Crown
15
4
3
2500
Amulet of Wisdom
30
3
0.5
1000
Phoenix Feather
0
5
0.05
5000
Frostbite Sword
85
4
8
2200
Elixir of Eternal Youth
0
5
0.2
30000
Whispering Amulet
20
4
0.5
3000
Ethereal Bow
60
5
3.2
7000
Mjolnir
302
5
88
16000
Dragonbone Greatsword
120
4
27
1500
Golden Compass of the Lost Isles
15
3
1.3
2700
Excalibur
235
5
40
15000